src/app-wrapper/metadata-helpers/merge-utils.ts

```diff
@@ -0,0 +1,73 @@
+import deepEqual from 'deep-equal'
+import type { MetadataStoreItem } from './types'
+
+// Helper to check if a value is considered "empty" for merge logic
+function isEmpty(value: unknown): boolean {
+    return (
+        value === null ||
+        value === undefined ||
+        value === '' ||
+        (Array.isArray(value) && value.length === 0) ||
+        (typeof value === 'object' &&
+            value !== null &&
+            !Array.isArray(value) &&
+            Object.keys(value).length === 0)
+    )
+}
+
+function isComplexFieldValue(value: unknown): boolean {
+    return typeof value === 'object' && value !== null
+}
+
+export function smartMergeWithChangeDetection(
+    existing: MetadataStoreItem | undefined,
+    newItem: MetadataStoreItem
+): { hasChanges: boolean; mergedItem: MetadataStoreItem } {
+    // If no existing item, just return the new item
+    if (!existing) {
+        return { hasChanges: true, mergedItem: newItem }
+    }
+
+    let hasChanges = false
+    const mergedItem = { ...existing } as Record<string, unknown>
```

Comment:
Would this be needed every time we "copy" an object?

Reply:
No, actually this was not needed. Well spotted.